File: PYTHON/binary_search.py
def binary_search(arr, target):
    left, right = 0, len(arr) - 1

    while left <= right:
        mid = left + (right - left) // 2

        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1

    return -1

if __name__ == "__main__":
    arr = list(map(int, input("Enter a sorted array (space-separated): ").split()))
    target = int(input("Enter the element to search for: "))

    result = binary_search(arr, target)

    if result != -1:
        print(f"Element {target} found at index {result}.")
    else:
        print(f"Element {target} not found in the array.")

File: C/Binary search.c
//BINARY SEARCH OF A SORTED ARRAY

#include <stdio.h>
int main() {
    int a[] = {3,4,7,9,10,14};
    int n=6,k=10;
    int low=0,high=n-1,pos=-1;
    while (low <= high) {
        int mid=low+(high-low)/2;
        if(a[mid]==k) {
            pos=mid;
            break; 
        }
        else if(a[mid]<k) {
            low=mid+1; 
        } 
        else 
        {
            high=mid-1; 
        }
    }

    if (pos!=-1) {
        printf("Element %d found at index %d\n", k,pos);
    } else {
        printf("Element %d not found in the array\n",k);
    }

    return 0;
}